src/core/dataset.ts

```diff
@@ -90,5 +101,23 @@ export class Dataset {
       return this;
     }
   }
+
+  public key(): (datum: any, index: number) => any;
+  /**
+   * Sets the key.
+   *
+   * @param { (d: any, i: number) => any} key
+   * @returns {Dataset} The calling Dataset.
+   */
+  public key(key: (datum: any, index: number) => any): Dataset;
+  public key(key?: (datum: any, index: number) => any): any {
```

Comment:
Agreed, that's probably better, and in keeping with the terminology here: http://bost.ocks.org/mike/constancy/